experiments/train_test_overlap/README.md

```diff
@@ -0,0 +1,88 @@
+# Train-Test Overlap Detection
+
+This directory tracks efforts for measuring train-test overlap in Marin datasets.
+
+## Directory Structure
+
+- `format/`: Contains scripts for converting datasets to JSONL format, which is required by the Dolma deduplication pipeline
+- `train_test/`: Contains scripts to run train-test overlap detection for each dataset of interest
+
+## Overview
+
+The process involves two main steps:
+
+1. Converting datasets to JSONL format using scripts in the `format/` directory, since this is the required format for Dolma's deduplication functionality
+
+2. Running train-test overlap detection using scripts in `train_test/` directory to identify and measure overlap between training and test sets for each dataset
+
+The overlap detection helps ensure data quality by identifying potential contamination between training and evaluation data.
+
+## Workflow Process
+
+This section outlines the complete workflow for running train-test overlap detection:
+
+### 1. Convert Raw Data to Dolma Format
+
+First, run `eval_datasets_overlap.py` to convert raw Hugging Face datasets to Dolma format suitable for deduplication:
+
+```bash
+python experiments/train_test_overlap/eval_datasets_overlap.py
+```
+
+This script:
+- Downloads evaluation datasets (MMLU, GSM8K, Math, TruthfulQA, BBH) from Hugging Face
+- Converts each dataset to the "decontamination" format with proper text fields for overlap detection
+
+### 2. Convert to Scenario Format and Consolidate
+
+Next, run `test_scenario_conversation.py` to convert the Dolma-formatted data to scenario files, and then use `consolidate_scenario_jsonl.py` to combine them into a single file:
+
+```bash
+# Run conversion only
+python experiments/train_test_overlap/format/test_scenario_conversation.py
+
+# Run consolidation (this will automatically run conversion step first if needed)
+python experiments/train_test_overlap/format/consolidate_scenario_jsonl.py
+```
+
+Alternatively, you can run the entire pipeline at once:
+
+```bash
+python experiments/train_test_overlap/format/run_scenario_pipeline.py
```

Comment:
Can we just give this as the only option to keep things simple.  Can have flags if we want to customize?